tests/test_routes.py

```diff
@@ -233,6 +252,64 @@ def test_authorized_project_member(self):
             r = c.get('/texts/urn:cts:formulae:andecavensis.form001/passage/2', follow_redirects=True)
             self.assertTemplateUsed('main::multipassage.html')
             self.assertIn('FORMULA ANDECAVENSIS 1', r.get_data(as_text=True))
+            # Make sure that a text that has no edition will throw an error
+            r = c.get('/texts/urn:cts:formulae:andecavensis.form003/passage/1', follow_redirects=True)
+            self.assertTemplateUsed("errors::unknown_collection.html")
+            self.assertIn('Angers 3.1' + _(' hat keine Edition.'), r.get_data(as_text=True))
+            c.get('/viewer/embedded/urn:cts:formulae:andecavensis.form001.lat001/0', follow_redirects=True)
+            self.assertTemplateUsed('viewer::multiviewer.html')
+            c.get('/viewer/embedded/urn:cts:formulae:andecavensis.form002.lat001/0', follow_redirects=True)
+            self.assertTemplateUsed('viewer::multiviewermirador.html')
+            c.get('/viewer/urn:cts:formulae:andecavensis.form001.lat001/0', follow_redirects=True)
+            self.assertTemplateUsed('viewer::newtabviewer.html')
+            c.get('/viewer/urn:cts:formulae:andecavensis.form002.lat001/0', follow_redirects=True)
+            self.assertTemplateUsed('viewer::miradorviewer.html')
+            c.get('/viewer/embedded/urn:cts:formulae:andecavensis.form001.lat001/abz', follow_redirects=True)
+            self.assertMessageFlashed('There is not an images for this formula. Showing the first page.')
```

Comment:
All of your flashed messages here will need to be changed to match the changes in the files above.